data/pigui/modules/saveloadgame.lua

```diff
@@ -197,42 +198,107 @@ end
 
 --=============================================================================
 
-local function makeEntryForSave(file)
-	local compatible, saveInfo = pcall(Game.SaveGameStats, file.name)
-	if not compatible then
-		saveInfo = {}
+-- Event callback once the savegame information has been loaded
+-- Updates the entryCache with the full details of the savegame.
+function SaveLoadWindow:onSaveGameStats(saveInfo)
+	print('onSaveGameStats : filename =', saveInfo.filename)
+	print('onSaveGameStats : compatible =', saveInfo.compatible)
+	print('onSaveGameStats : gameTime =', saveInfo.time)
+	print('onSaveGameStats : system =', saveInfo.system)
+
+	if not self.entryCache then
+		print('onSaveGameStats: self.entryCache is NIL!!')
+		return
 	end
 
-	local location = saveInfo.system or lc.UNKNOWN
+	if not self.entryCache[saveInfo.filename] then
+		print('onSaveGameStats: self.entryCache['.. saveInfo.filename .. '] is NIL!!')
+		return
+	end
 
+	local location = saveInfo.system or lc.UNKNOWN
 	if saveInfo.docked_at then
 		location = location .. ", " .. saveInfo.docked_at
 	end
 
-	local saveEntry = SaveGameEntry:clone({
-		name = file.name,
-		compatible = compatible,
-		isAutosave = file.name:sub(1, 1) == "_",
-		character = saveInfo.character,
-		timestamp = file.mtime.timestamp,
-		gameTime = saveInfo.time,
-		duration = saveInfo.duration,
-		locationName = location,
-		credits = saveInfo.credits,
-		shipName = saveInfo.shipName,
-		shipHull = saveInfo.shipHull,
-	})
+	-- local saveEntry = self:getSaveEntry(saveInfo.filename)
+	-- saveEntry.compatible = saveInfo.compatible
+	-- saveEntry.gameTime = saveInfo.time
+	-- saveEntry.locationName = location
+	-- self.entryCache[saveInfo.filename] = saveEntry
 
 	-- Old saves store only the name of the ship's *model* file for some dumb reason
 	-- Treat the model name as the ship id and otherwise ignore it if we have proper data
+	local shipHull
 	if not saveInfo.shipHull then
 		local shipDef = ShipDef[saveInfo.ship]
-
 		if shipDef then
-			saveEntry.shipHull = shipDef.name
+			shipHull = shipDef.name
 		end
+	else
+		shipHull = saveInfo.shipHull
 	end
 
+	self.entryCache[saveInfo.filename].character = saveInfo.character
+	self.entryCache[saveInfo.filename].compatible = saveInfo.compatible
+	self.entryCache[saveInfo.filename].credits = saveInfo.credits
+	self.entryCache[saveInfo.filename].duration = saveInfo.duration
+	self.entryCache[saveInfo.filename].gameTime = saveInfo.time
+	self.entryCache[saveInfo.filename].locationName = location
+	self.entryCache[saveInfo.filename].shipName = saveInfo.shipName
+	self.entryCache[saveInfo.filename].shipHull = shipHull
```

Comment:
Note: Lua cannot optimize out the repeated table lookups here. Prefer storing `self.entryCache[saveInfo.filename]` in a local variable and directly assigning to it.

Reply:
Heh, ok, thanks for the heads-up.  Currently working with someone who very much dislikes "unnecessary" local variables, and C just optimises this sort of stuff out.

Reply:
Fixed in c32dd13852548473bb5ce72aabf3c54ec7571008